device_doctor/lib/src/ios_device.dart

```diff
@@ -76,6 +78,39 @@ class IosDeviceDiscovery implements DeviceDiscovery {
     }
     return healthCheckResult;
   }
+
+  Future<HealthCheckResult> certCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
+      final String certCheckResult =
+          await eval('security', <String>['find-identity', '-p', 'codesigning', '-v'], processManager: processManager);
+      if (certCheckResult.contains('Apple Development: Flutter Devicelab') &&
+          certCheckResult.contains('1 valid identities found')) {
+        healthCheckResult = HealthCheckResult.success(kCertCheckKey);
+      } else {
+        healthCheckResult = HealthCheckResult.failure(kCertCheckKey, certCheckResult);
+      }
+    } on BuildFailedError catch (error) {
```

Comment:
What if the command execution fails? would this catch that error?

Reply:
both the command execution failure and the exitcode not equal to 0 will be returned as BuildFailedError.